spec/index.bs

```diff
@@ -1922,6 +1947,68 @@ Issue: `GPUDevice` doesn't really need the cross-origin policy restriction.
 It should be usable from multiple agents regardless. Once we describe the serialization
 of buffers, textures, and queues - the COOP+COEP logic should be moved in there.
 
+## Example ## {#initialization-examples}
+
+<div class="example">
+    A more robust example of requesting a {{GPUAdapter}} and {{GPUDevice}} with error handling:
+    <pre highlight="js">
+        let gpuDevice = null;
+
+        async function initializeWebGPU() {
+            // Check to ensure the user agent supports WebGPU.
+            if (!('gpu' in navigator)) {
+                console.error('User agent doesn't support WebGPU.');
+                return false;
+            }
+
+            // Request an adapter.
+            const gpuAdapter = await navigator.gpu.requestAdapter();
+
+            // requestAdapter may resolve with null if no suitable adapters are found.
+            if (!gpuAdapter) {
+                console.error('No WebGPU adapters found.');
+                return false;
+            }
+
+            // Request a device.
+            // Note that the promise will reject if invalid options are passed to the optional
+            // dictionary. To avoid the promise rejecting always check any features and limits
+            // against the adapters features and limits prior to calling requestDevice().
+            gpuDevice = await gpuAdapter.requestDevice();
+
+            // requestDevice will never return null, but if a valid device request can't be
+            // fulfilled for some reason it may resolve to a device which has already been lost.
+            // Additionally, devices can be lost at any time after creation, usually for external
+            // reasons (ie: driver updates), so it's a good idea to always handle lost devices
```

Comment:
nit: more common examples to convey that device loss is realistic. (I'd like to say "GPU resource reclamation" but realistically not sure whether browsers will actually do that)

```suggestion
            // reasons (e.g. driver reset or eGPU unplug), so it's a good idea to always handle lost devices
```

Reply:
I think browser resource management is a good scenario to include, even if it's a hypothetical. It conveys a good sense of "something that can happen during everyday use" rather than "something that only happens in exceptional circumstances or with weird hardware configurations."